com.unity.render-pipelines.high-definition/Editor/VFXGraph/VFXHDRPSubTarget.cs

```diff
@@ -0,0 +1,374 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using UnityEngine;
+using UnityEditor.ShaderGraph;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEditor.VFX;
+
+namespace UnityEditor.Rendering.HighDefinition.ShaderGraph
+{
+    [InitializeOnLoad]
+    static class VFXHDRPSubTarget
+    {
+        internal const string Inspector = "Rendering.HighDefinition.VFXShaderGraphGUI";
+
+        static VFXHDRPSubTarget()
+        {
+            VFXSubTarget.OnPostProcessSubShader += PostProcessSubShader;
```

Comment:
I realized this could be a future design issue : OnPostProcessSubShader & VFXHDRPSubTarget are static, the OnPostProcessSubShader is appended with first C# compilation.
HDRP & URP can already cohabit within the same project (but there isn't any VFXURPSubTarget)
So either we should register/unregister this callback or simply handle a verification in "PostProcessSubShader" (the second option is probably the simplest).
It can be done later and this is maybe something you already anticipated since you added this comment :
`https://github.com/Unity-Technologies/Graphics/pull/3467/files#diff-42390dfcfd4234620ef9565c7a776d24cd6f05e33c1ddc99da21165533a4cc38R42`
Not really an issue yet.